packages/data-context/test/unit/sources/GitDataSource.spec.ts

```diff
@@ -99,6 +99,64 @@ describe('GitDataSource', () => {
     expect(modified.lastModifiedTimestamp).not.to.be.undefined
   })
 
+  it(`handles files with special characters on ${os.platform()}`, async () => {
```

Comment:
Suggest adding a comment linking to to https://github.com/cypress-io/cypress/issues/22454 in this test. I love when tests link issues so I can tell exactly what bug they're protecting against. :)